### DISCUSSION TITLE
DiffPool example what is mask in TUDataset

### DISCUSSION THREAD

**Discussion by lisiq** [pyg-team/pytorch_geometric#4564] [Q&A] [answered]
Hi everyone, I was trying to understand the code and the dataset in [pytorch_geometric examples](
https://github.com/pyg-team/pytorch_geometric/blob/master/examples/proteins_diff_pool.py). I don't understand how is the mask in PROTEINS dataset generated, what does it represent and how is DiffPool making use of it?

Thank you very much!

**Comment by rusty1s** [comment, marked as answer]
The `mask` is generated by the `ToDense` transform, and indicates which nodes are valid nodes and which nodes were added due to padding reasons in order to create a dense adjacency matrix of shape `[batch_size, max_num_nodes, max_num_nodes]`. As such, `mask` is a `[batch_size, max_num_nodes]` matrix.

**Reply by lisiq** [reply]
Thank you very much that was helpful!

**Reply by lisiq** [reply]
@rusty1s  Is there a tutorial available on how to create these datasets?

**Reply by rusty1s** [reply]
What do you mean exactly? [This](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html) and [this](https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html) tutorial might be of interest to you!

**Reply by lisiq** [reply]
I have some graphs with different number of nodes, and I was wondering how to create a similar dataset. I found this tutorial https://colab.research.google.com/github/AntonioLonga/PytorchGeometricTutorial/blob/main/Tutorial15/Tutorial15.ipynb#scrollTo=69a46b4c so now I created similar textfiles and it seems to have solved my problem. Thank you!
